java-squid/src/main/java/org/sonar/java/se/symbolicvalues/SymbolicValueRelation.java

```diff
@@ -28,20 +32,35 @@
 import java.util.List;
 import java.util.Set;
 
-public abstract class SymbolicValueRelation {
+public abstract class SymbolicValueRelation extends BinarySymbolicValue {
 
-  protected final SymbolicValue v1;
-  protected final SymbolicValue v2;
+  public SymbolicValueRelation(int id) {
+    super(id);
+  }
 
   protected SymbolicValueRelation(SymbolicValue v1, SymbolicValue v2) {
-    this.v1 = v1;
-    this.v2 = v2;
+    super(-1);
```

Comment:
As discussed every object created this way is NOT a symbolic value per say. As such I would expect at the very least some invariants to prevent any kind of SVrelation to leak on the stack.